metrics_sdk/lib/opentelemetry/sdk/metrics/export/periodic_metric_reader.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Export
+        # PeriodicMetricReader provides a minimal example implementation.
+        class PeriodicMetricReader < MetricReader
+          def initialize(export_interval_millis: Float(ENV.fetch('OTEL_METRIC_EXPORT_INTERVAL', 60_000)),
+                         export_timeout_millis: Float(ENV.fetch('OTEL_METRIC_EXPORT_TIMEOUT', 30_000)),
+                         exporter: nil)
+            super()
+
+            @export_interval = export_interval_millis / 1000.0
+            @export_timeout = export_timeout_millis / 1000.0
+            @exporter = exporter
+            @thread   = nil
+            @continue = false
+
+            start
+          end
+
+          def start
+            @continue = true
+            if @exporter.nil?
+              OpenTelemetry.logger.warn 'Missing exporter in PeriodicMetricReader.'
+            elsif @thread&.alive?
+              OpenTelemetry.logger.warn 'PeriodicMetricReader is still running. Please close it if it needs to restart.'
+            else
+              @thread = Thread.new do
+                while @continue
```

Comment:
While running the periodic exporting metric reader with the OTLP exporter, I hit an error when there are no metrics to send in a given period. 

```
* E, [2024-07-31T14:57:16.719070 #57663] ERROR -- : OpenTelemetry error: unexpected error in 
OTLP::MetricsExporter#encode - undefined method `group_by' for nil:NilClass`
```

I believe the BatchSpanProcessor works around this by using ConditionVariable objects to skip the export unless the batch has enough objects. I think it'll be resolved by #1627. We may want to wait to ship the periodic exporting metric reader in a version of the metrics_sdk until this refactor happens.

Reply:
I added the mechanism that check that if metrics is empty then don't export. I didn't use batch since I think periodic exporting is different than batch exporting (but please correct me if I am wrong).

Also updated the thread improvement part based on batch processor

Reply:
Thank you! Also, I don't think we need to include batch size. I like the documentation updates too 🙂